sui_core/src/execution_engine.rs

```diff
@@ -95,13 +96,14 @@ fn execute_transaction<S: BackingPackageStore>(
             SingleTransactionKind::Call(c) => {
                 let mut inputs: Vec<_> = object_input_iter.by_ref().take(input_size).collect();
                 // unwraps here are safe because we built `inputs`
-                let package = inputs.pop().unwrap();
+                // TODO don't load and push the package object here in the first place
```

Comment:
The only reason we read the package before is to make sure all objects used by this transaction (in this case, the package of the Move call) exists on-chain.
Perhaps we could optimize that by checking the VM cache. Not sure.

Reply:
The VM will error if the module doesn't exist, but yeah, you might want to check that very early on